src/renderer/components/GeneralSettings/index.tsx

```diff
@@ -34,8 +34,23 @@ function InstallPathSettingItem(props: { path: string, setPath: (path: string) =
     );
 }
 
+const DisableWarningSettingItem = (props: {disableWarning: boolean, setDisableWarning: CallableFunction}) => {
+    const handleClick = () => {
+        props.setDisableWarning(!props.disableWarning);
+        settings.set('mainSettings.disableWarning', !props.disableWarning);
```

Comment:
Race condition.

Setting `disableWarning` first and then using it again. Use a temporary variable.

```suggestion
        const newState = !props.disableWarning;
        props.setDisableWarning(newState);
        settings.set('mainSettings.disableWarning', newState);
```